core/src/test/scala/unit/kafka/server/TierStateMachineTest.scala

```diff
@@ -153,12 +157,13 @@ class ReplicaFetcherTierStateMachineTest {
     assertEquals(11L, replicaState.logEndOffset)
   }
 
-  @Test
-  def testFencedOffsetResetAfterMovedToRemoteTier(): Unit = {
+  @ParameterizedTest
+  @ArgumentsSource(classOf[TierStateMachineTest.Params])
+  def testFencedOffsetResetAfterMovedToRemoteTier(truncateOnFetch: Boolean, useFutureLog: Boolean): Unit = {
     val partition = new TopicPartition("topic", 0)
     var isErrorHandled = false
     val mockLeaderEndpoint = new MockLeaderEndPoint(truncateOnFetch = truncateOnFetch, version = version)
-    val mockTierStateMachine = new MockTierStateMachine(mockLeaderEndpoint) {
+    val mockTierStateMachine = new MockTierStateMachine(mockLeaderEndpoint, useFutureLog) {
```

Comment:
Given:

a. `MockTierStateMachine` only refers to `useFutureLog` to carry it over to the `TierStateMachine`;
b. `TierStateMachine ` only refers to `useFutureLog ` in `buildRemoteLogAuxState`;
c. `buildRemoteLogAuxState` is only invoked from `TierStateMachine#start`;
d. `MockTierStateMachine` overrides `start` and does not call `super.start()`;

It seems having `useFutureLog` as a test param is unnecessary?

Reply:
Unfortunately, you're correct! Removed the `useFutureLog`. I'll think about how to do unit test for it. Thanks